pkg/odo/cli/list/services/services.go

```diff
@@ -0,0 +1,198 @@
+	o.clientset = clientset
+}
+
+func (o *ServiceListOptions) Complete(cmdline cmdline.Cmdline, _ []string) error {
+	var err error
+	o.contextDir, err = os.Getwd()
+	if err != nil {
+		return err
+	}
+
+	o.Context, err = genericclioptions.New(genericclioptions.NewCreateParameters(cmdline))
+	if err != nil {
+		return err
+	}
+
+	if o.namespaceFlag == "" && !o.allNamespacesFlag {
+		o.namespaceFlag = o.GetProject()
+	}
+
+	return err
+}
+
+func (o *ServiceListOptions) Validate() error {
+	if o.allNamespacesFlag && o.namespaceFlag != "" {
+		return errors.New("cannot use --all-namespaces and --namespace flags together")
+	}
+	return nil
+}
+
+func (o *ServiceListOptions) Run(_ context.Context) error {
+	var s *log.Status
+	if o.allNamespacesFlag {
+		s = log.Spinner("Listing bindable services from all namespaces")
+		defer s.End(false)
+	} else {
+		s = log.Spinner(fmt.Sprintf("Listing bindable services from namespace %q", o.namespaceFlag))
+		defer s.End(false)
+	}
+	services, err := o.run()
+	if err != nil {
+		return err
+	}
+	s.End(true)
+	HumanReadable(services)
+	return nil
+}
+
+func (o *ServiceListOptions) run() ([]unstructured.Unstructured, error) {
+	services := []unstructured.Unstructured{}
+
+	if o.allNamespacesFlag {
+		projects, err := o.clientset.ProjectClient.List()
+		if err != nil {
+			return services, err
+		}
+		for _, project := range projects.Items {
+			svcs, err := o.clientset.BindingClient.GetServiceInstances(project.Name)
+			if err != nil {
+				return services, err
+			}
+			for k := range svcs {
+				services = append(services, svcs[k])
+			}
+		}
+	} else {
+		svcs, err := o.clientset.BindingClient.GetServiceInstances(o.namespaceFlag)
+		if err != nil {
+			return services, err
+		}
+		for k := range svcs {
+			services = append(services, svcs[k])
+		}
+	}
+	return services, nil
+
+}
+
+func (o *ServiceListOptions) RunForJsonOutput(_ context.Context) (out interface{}, err error) {
+	services, err := o.run()
+	if err != nil {
+		return nil, err
+	}
+
+	var servicesList []api.BindableService
+	for _, svc := range services {
+		servicesList = append(servicesList, api.BindableService{Name: svc.GetName(), Namespace: svc.GetNamespace()})
+	}
+	return map[string]interface{}{"bindableServices": servicesList}, nil
+}
+
+func HumanReadable(services []unstructured.Unstructured) {
+	if isServiceSliceEmpty(services) {
+		log.Error("no bindable Operator backed services found")
+		return
+	}
+	fmt.Println()
+	t := ui.NewTable()
+	t.AppendHeader(table.Row{"NAME", "NAMESPACE"})
+	for _, svc := range services {
+		t.AppendRow(table.Row{fmt.Sprintf("%s/%s.%s", svc.GetName(), svc.GetKind(), strings.Split(svc.GetAPIVersion(), "/")[0]), svc.GetNamespace()})
+	}
+	t.Render()
+}
+
+func isServiceSliceEmpty(services []unstructured.Unstructured) bool {
+	if len(services) == 0 {
+		return true
+	}
+	if len(services) == 1 {
+		if reflect.DeepEqual(services[0], unstructured.Unstructured{}) {
+			return true
+		}
+	}
```

Comment:
In which circumstances can we have this case?

Reply:
I still don't understand how the list could contain a single "zero" Unstructured element. When would it be added?

Reply:
I take it back. TBH, I found it weird too. Anyway, it should be fixed now.